### ISSUE TITLE
Tasks Not Disappearing

### ISSUE THREAD

**Issue by ksinghal** [firebase/firebase-queue#67]
I am calling the resolve function, but the task seems to be reclaimed instantly by the worker even after it has already been completed. In other words, the tasks never complete - they keep happening over and over infinitely... I have added a screenshot of my spec if that helps
<img width="712" alt="screen shot 2016-05-26 at 7 24 19 am" src="https://cloud.githubusercontent.com/assets/4765829/15573155/f0b74680-2312-11e6-9bf9-815aad081dc6.png">

**Comment by ksinghal** [comment]
I don't think this is the same as #48, because I do not have any indexes in my firebase. However, my problem is exactly the sync vs async issue described in this comment: https://github.com/firebase/firebase-queue/issues/48#issuecomment-208400101

I am unable to resolve tasks async, only sync.

**Comment by katowulf** [comment]
Please provide a minimal repro and also sample data including what the task event looks like at each point in the process (not just the spec).

**Comment by katowulf** [comment]
Also, hi!! Welcom!! ☃☃☃

**Comment by ksinghal** [comment]
Hit Katowulf, thank you for your prompt response! Here is an example of server side NodeJS code that reproduces this issue. I am using node `v5.5.0`, and using firebase `2.4.2`, and have tried using firebase-queue `1.3.0`, `1.3.1`, and `1.4.0`

```
var express = require('express');
var Queue = require('firebase-queue'), Firebase = require('firebase');
var app = express();
app.set('port', (process.env.PORT || 5000));
app.use(express.static(__dirname + '/public'));

app.get('/', function(request, response) {
  response.send('Hello World!')
});

app.listen(app.get('port'), function() {
  console.log("Node app is running at localhost:" + app.get('port'))
});


var ref = new Firebase('https://spotlight-dev.firebaseio.com/');

var queue = new Queue(ref.child('queues').child('create_reservation'), {'numWorkers' : 1, 'specId' : 'default_spec'}, function(data, progress, resolve, reject) {
	// Read and process task data
	setTimeout(function() { resolve(data); }, 3000);
});
```

The task goes between the two states in the attached images:
![screen shot 2016-05-26 at 1 10 11 pm](https://cloud.githubusercontent.com/assets/4765829/15583340/5b828aa8-2343-11e6-86b9-93b86ee20cf1.png)
![screen shot 2016-05-26 at 1 10 21 pm](https://cloud.githubusercontent.com/assets/4765829/15583344/5d76f7e0-2343-11e6-92b7-7a535b0cc599.png)

**Comment by drtriumph** [comment]
Could you include your specs and share some of the code that adds onto the queue. Also, is the express stuff important to the repro?

**Comment by ksinghal** [comment]
Here is an image of an example spec that exhibits the behavior described. The tasks are attached to the queue, the spec worker processes them, and finally the `_state`, `_progress`, and `_owner` are removed and then added again. The single spec worker in the image keeps running each task over and over again and i have never seen it stop so I assume it's infinite

I have also updated my comment above to remove the unnecessary ExpressJS code

<img width="570" alt="screen shot 2016-05-28 at 4 02 16 pm" src="https://cloud.githubusercontent.com/assets/4765829/15629692/9412f4b0-24ed-11e6-9072-f4cffe92da6a.png">

**Comment by ksinghal** [comment]
Hi, I was wondering if you needed any more information on my end to get to the bottom of this issue?

**Comment by gauravk92** [comment]
+1

**Comment by amartone** [comment]
+1

**Comment by ksinghal** [comment]
Hello Firebase Team,

This issue had resolved itself for a period of a month or so, after upgrading to firebase 3.x and the new firebase queue. Now after 3 and a half months of it working properly with asynchronous resolves, and with zero code changes on my end it has reappeared today unexpectedly. I am fairly frustrated and confused as I am not sure what is leading to this issue. It seems to pop up unexpectedly and not due to changes on my end

**Comment by ksinghal** [comment]
UPDATE:

Now I am getting worried, because after popping up for a period of 2 hours, the issue resolved itself again with no modification of my system's code. As I am hoping to put my applications into production soon, I really do not want to risk a re-ocurrence of this issue without a known cause or way to mitigate it. Thank you again for your continued support, I look forward to hearing back

**Comment by prescottprue** [comment]
@ksinghal Kinda a long shot question here, but did you redeploy your code at all? If it was reinstalled without a specific enough (using `^`) package.json file you might be getting a different version of `firebase` or another dependency then you had before. I had this happen when changing absolutely 0 code and pushing just for a change in a README.

Example:
Your package file has something like `^3.0.0` and everything was working before on 3, as soon as a patch is released and you reinstall you will be on that changed version (maybe 3.0.1).

**Comment by ksinghal** [comment]
I simply restarted the server locally, without redeploying and without running "npm install"

**Comment by ksinghal** [comment]
to clarify, I restarted the server many times without making modifications, and the issue was still present. Then, all of a sudden after a few hours the issue just seemed to stop happenning

**Comment by katowulf** [comment]
Possibly related to #48.
